### ISSUE TITLE
AMP-Social Share svg ignore color attribute

### ISSUE THREAD

**Issue by merakmarey** [ampproject/amphtml#26497]
According to https://amp.dev/documentation/examples/components/amp-social-share/?format=websites, icons colors can be styled using a css color attribute, same as the background-color attribute, however on the invoke of registerElement in amp-social-share-0.1.js the svg fill attribute of the default icons is fixed to '#fff' making the color attribute  useless.

@paul-matthews

**Comment by caroqliu** [comment]
Thanks for filing this issue. It is possible to override the default styles using the [SVG data](https://github.com/ampproject/amphtml/blob/master/third_party/optimized-svg-icons/amp-social-share-svgs.css).

Here is a sample with a non-white fill for the icon through custom styling:
```html
<!doctype html>
<html ⚡>
<head>
  <meta charset="utf-8">
  <title>My AMP Page</title>
  <link rel="canonical" href="self.html" />
  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-social-share" src="https://cdn.ampproject.org/v0/amp-social-share-0.1.js"></script>
<style amp-custom>
  /* a custom sharing icon */
  amp-social-share.custom-style {
    border: 1px solid #ff0000;
    background-color: #ffffff;
    background-image: url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20512%20512%22%3E%3Cpath%20fill%3D%22%23ff0000%22%20d%3D%22M186.4%20142.4c0%2019-15.3%2034.5-34.2%2034.5%20-18.9%200-34.2-15.4-34.2-34.5%200-19%2015.3-34.5%2034.2-34.5C171.1%20107.9%20186.4%20123.4%20186.4%20142.4zM181.4%20201.3h-57.8V388.1h57.8V201.3zM273.8%20201.3h-55.4V388.1h55.4c0%200%200-69.3%200-98%200-26.3%2012.1-41.9%2035.2-41.9%2021.3%200%2031.5%2015%2031.5%2041.9%200%2026.9%200%2098%200%2098h57.5c0%200%200-68.2%200-118.3%200-50-28.3-74.2-68-74.2%20-39.6%200-56.3%2030.9-56.3%2030.9v-25.2H273.8z%22%2F%3E%3C%2Fsvg%3E');
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
  }
</style>
</head>
<body>
  <amp-social-share type="linkedin"
    class="custom-style">
  </amp-social-share>
</body>
</html>
```

Closing the issue for now, feel free to re-open.

**Comment by merakmarey** [comment]
So, basically in order to have style applied to this icon I would need to recreate the whole svg?...wow...

**Comment by wassgha** [comment]
We have three options here:
- With the current implementation, we can only change the color is either by replacing the `background-image` or by using a filter (for example `filter: brightness(0.2) sepia(1) hue-rotate(180deg) saturate(5);` for a blue tone).
- We can expose the `<svg>` as a child instead of a `background-image` which will allow developers to use `color` and `fill` on it but that's a breaking change and requires a version bump
- We can add a `color` attribute to the component and that requires setting the `background-image` on the javascript side

**Comment by merakmarey** [comment]
If we replace the fill color value in the string passed to the registerElement call...that would affect all the amp-social-share elements in the page?..btw, is not registerElement deprecated?

**Comment by merakmarey** [comment]
Don't really understand why the background-image has any relation with the svg... can we just replace the fill attribute value with the color value of the element at the moment of the element creation?

**Comment by wassgha** [comment]
It's not as straightforward because the the svg is currently added as a `background-image` (not as a child `<svg>` element). If we change to the latter implementation, we'd break websites that used to rely on styling `background-image` (hence the need for a version bump).

**Comment by merakmarey** [comment]
Has anything changed with this or is still a no can do situation?

**Comment by caroqliu** [comment]
amp-social-share 1.0 allows for customizing the colors, but 0.1 does not: https://amp.dev/documentation/components/amp-social-share-v1.0/?format=websites#custom-styles

**Comment by merakmarey** [comment]
This does not work, not even with v1.0 

`<!doctype html>
<html ⚡>
<head>
  <meta charset="utf-8">
  <title>My AMP Page</title>
  <link rel="canonical" href="self.html" />
  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-social-share" src="https://cdn.ampproject.org/v0/amp-social-share-1.0.js"></script>
<style amp-custom>
  /* a custom sharing icon */
  
  amp-social-share[type='linkedin'] {
  color: blue;
  background: red;
}
  
</style>
</head>
<body>
  <amp-social-share type="linkedin"
    class="custom-style">
  </amp-social-share>
</body>
</html>`

**Comment by caroqliu** [comment]
You will need to enable the `"bento"` or `"bento-social-share"` experiment somehow. This is the [playground example](https://playground.amp.dev/#share=PCFkb2N0eXBlIGh0bWw+CjxodG1sIOKaoT4KPGhlYWQ+CiAgPG1ldGEgY2hhcnNldD0idXRmLTgiPgogIDx0aXRsZT5NeSBBTVAgUGFnZTwvdGl0bGU+CiAgPGxpbmsgcmVsPSJjYW5vbmljYWwiIGhyZWY9InNlbGYuaHRtbCIgLz4KICA8bWV0YSBuYW1lPSJ2aWV3cG9ydCIgY29udGVudD0id2lkdGg9ZGV2aWNlLXdpZHRoLG1pbmltdW0tc2NhbGU9MSxpbml0aWFsLXNjYWxlPTEiPgogIDxzdHlsZSBhbXAtYm9pbGVycGxhdGU+Ym9keXstd2Via2l0LWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoOy1tb3otYW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGg7LW1zLWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoO2FuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RofUAtd2Via2l0LWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1ALW1vei1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1tcy1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1vLWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1Aa2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fTwvc3R5bGU+PG5vc2NyaXB0PjxzdHlsZSBhbXAtYm9pbGVycGxhdGU+Ym9keXstd2Via2l0LWFuaW1hdGlvbjpub25lOy1tb3otYW5pbWF0aW9uOm5vbmU7LW1zLWFuaW1hdGlvbjpub25lO2FuaW1hdGlvbjpub25lfTwvc3R5bGU+PC9ub3NjcmlwdD4KICA8c2NyaXB0IGFzeW5jIHNyYz0iaHR0cHM6Ly9jZG4uYW1wcHJvamVjdC5vcmcvdjAuanMiPjwvc2NyaXB0PgogIDxzY3JpcHQgYXN5bmMgY3VzdG9tLWVsZW1lbnQ9ImFtcC1zb2NpYWwtc2hhcmUiIHNyYz0iaHR0cHM6Ly9jZG4uYW1wcHJvamVjdC5vcmcvdjAvYW1wLXNvY2lhbC1zaGFyZS0xLjAuanMiPjwvc2NyaXB0Pgo8c3R5bGUgYW1wLWN1c3RvbT4KICAvKiBhIGN1c3RvbSBzaGFyaW5nIGljb24gKi8KICAKICBhbXAtc29jaWFsLXNoYXJlW3R5cGU9ImxpbmtlZGluIl0gewogIGNvbG9yOiBibHVlOwogIGJhY2tncm91bmQ6IHJlZDsKfQogIAo8L3N0eWxlPgogIDxzY3JpcHQ+CiAgKHNlbGYuQU1QID0gc2VsZi5BTVAgfHwgW10pLnB1c2goZnVuY3Rpb24gKEFNUCkgewogICAgQU1QLnRvZ2dsZUV4cGVyaW1lbnQoJ2JlbnRvJywgdHJ1ZSk7CiAgfSk7Cjwvc2NyaXB0Pgo8L2hlYWQ+Cjxib2R5PgogIDxhbXAtc29jaWFsLXNoYXJlIHR5cGU9ImxpbmtlZGluIgogICAgY2xhc3M9ImN1c3RvbS1zdHlsZSI+CiAgPC9hbXAtc29jaWFsLXNoYXJlPgo8L2JvZHk+CjwvaHRtbD4=) using the [invalid AMP way to do it for development](https://amp.dev/documentation/guides-and-tutorials/start/bento_guide/). Here are [valid AMP ways to enable experiments](https://amp.dev/documentation/guides-and-tutorials/learn/experimental/).

**Comment by merakmarey** [comment]
So, basically, if I want to style, I have to forget about valid AMP...right?

**Comment by caroqliu** [comment]
> So, basically, if I want to style, I have to forget about valid AMP...right?

No, there are ways to enable experiments that are valid AMP, as documented in ["Enable Experimental Features"](https://amp.dev/documentation/guides-and-tutorials/learn/experimental/).

That said, the 1.0 version of amp-social-share will be planned to launch as valid AMP soon. I'll update this issue when that happens. Thank you!

**Comment by merakmarey** [comment]
Thanks to you! Ill wait! :)
